New Animals

Azaysol opened the pull request:
new animal update, lots of changes sending over to start review
-still pending sounds will send another update hopefully tomorrow with finalized sounds


closes #1357

Azaysol committed:
added Direwolf

added Direwolf

Azaysol committed:
WIP Hare

WIP Hare

Azaysol committed:
WIP Hog

Azaysol committed:
Merge remote-tracking branch 'origin/new-animals' into new-animals

# Conflicts:
#	src/main/java/net/dries007/tfc/client/ClientEvents.java
#	src/main/java/net/dries007/tfc/objects/entity/EntitiesTFC.java

Azaysol committed:
WIP Zebu

Azaysol committed:
WIP SeaLion

Azaysol committed:
WIP Gazelle

Azaysol committed:
Reorganizing texture locations

Azaysol committed:
WIP Wildebeest

Azaysol committed:
WIP Quail

Azaysol committed:
WIP Grouse

Azaysol committed:
WIP Mongoose

Azaysol committed:
WIP Turkey

Azaysol committed:
WIP Jackal

Azaysol committed:
WIP MuskOx

Azaysol committed:
WIP Yak

Azaysol committed:
WIP BlackBear

Azaysol committed:
WIP Cougar

Azaysol committed:
WIP Coyote

Azaysol committed:
Renamed Bear to Grizzly Bear

Azaysol committed:
adding Tabula credits and fixing some copy pasta

Azaysol committed:
updating animal spawn locations

Azaysol committed:
Lot of fixes for animal rendering issues and floating babies and some small tweaks to new animals

Azaysol committed:
Update Models for Zebu and Boar

Azaysol committed:
Textures for Boar, Turkey, Zebu and Gazelle plus some tweaks

Azaysol committed:
Mongoose texture from EER and some tweaks

Azaysol committed:
Direwolf and MuskOx textures and updates. some fixes for Turkey and Zebu

Azaysol committed:
Updated Wildebeest

Azaysol committed:
Updated Yak and some changes to Wildebeest

Azaysol committed:
Updated Quail Model and Textures

eerussianguy committed:
deleted seal, fixed all loot tables

Azaysol committed:
Merge pull request #1 from eerussianguy/new-animals

loot table fixes + seal deletion

eerussianguy committed:
sounds, lang finishes, egg colors

eerussianguy committed:
configs, finished lang

eerussianguy committed:
small fix

Azaysol committed:
Merge pull request #2 from eerussianguy/new-animals

sounds, lang finishes, configs and egg colors

Azaysol committed:
Update to Coyote and Jackals, removed unused sealion

Azaysol committed:
Merge remote-tracking branch 'origin/new-animals' into new-animals

Azaysol committed:
Update to Cougar and Panther

Azaysol committed:
Update for Grouse textures

Azaysol committed:
Update for Hare textures

Azaysol committed:
Lots of fixes for animations, models, textures

Azaysol committed:
Lots of fixes for hitboxes, sizes, loot tables and some balances

Azaysol committed:
Lots of config changes, ratios and balance done for animals

Azaysol committed:
some fixes and updates

Azaysol committed:
Merge pull request #4 from TerraFirmaCraft/1.12.x

rebase

Azaysol committed:
Merge remote-tracking branch 'remotes/origin/1.12.x' into new.animals

# Conflicts:
#	src/main/java/net/dries007/tfc/ConfigTFC.java
#	src/main/resources/assets/tfc/loot_tables/animals/chicken.json
#	src/main/resources/assets/tfc/loot_tables/animals/duck.json
#	src/main/resources/assets/tfc/loot_tables/animals/pheasant.json
#	src/main/resources/assets/tfc/loot_tables/animals/rabbit.json
#	src/main/resources/assets/tfc/loot_tables/animals/squid.json
#	src/main/resources/assets/tfc/loot_tables/animals/wolf.json

Azaysol committed:
finalizing updates and fixes plus code cleanup

Azaysol committed:
Sound updates and additions

Azaysol commented:
sounds have been added, everything should be complete now

DisasterMoo commented on src/main/java/net/dries007/tfc/client/ClientEvents.java:
```diff
@@ -215,6 +231,7 @@ else if (mc.world.provider.getDimension() == 0)
 
                     if (chunkDataValid)
                     {
+                        list.add(GRAY + "Rainfall: " + WHITE + data.getRainfall());
```
Probably missed from earlier, this was moved to not require the debug enabled, so leaving this line here will most certainly duplicate F3 menu.

Azaysol replied:
unintended, changed as requested

DisasterMoo commented on src/main/java/net/dries007/tfc/objects/entity/animal/EntityAlpacaTFC.java:
```diff
@@ -124,15 +147,14 @@ protected SoundEvent getAmbientSound()
-    protected void playStepSound(BlockPos pos, Block blockIn)
-    {
-        playSound(TFCSounds.ANIMAL_ALPACA_STEP, 0.15F, 1.0F);
-    }
```
Intentional change? Wasn't `TFCSounds.ANIMAL_ALPACA_STEP` correct?

Azaysol replied:
intentional change

DisasterMoo commented on src/main/java/net/dries007/tfc/objects/entity/animal/EntityCougarTFC.java:
```diff
@@ -0,0 +1,195 @@
+    @Override
+    protected void playStepSound(BlockPos pos, Block blockIn)
+    {
+        this.playSound(SoundEvents.ENTITY_HORSE_STEP, 0.15F, 1.0F); // Close enough
```
Are you certain that the horse step sound is close enough for a cougar?

Azaysol replied:
thats a good point, ill add something for the feline steps

DisasterMoo commented on src/main/java/net/dries007/tfc/objects/entity/animal/EntityHareTFC.java:
```diff
@@ -0,0 +1,567 @@
+import net.dries007.tfc.world.classic.biomes.BiomesTFC;
+
+@ParametersAreNonnullByDefault
+public class EntityHareTFC extends EntityAnimalMammal implements IHuntable
```
This one should probably extend rabbit class so you can cleanup a lot of copy-pasta below

Azaysol replied:
changes made

DisasterMoo commented on src/main/java/net/dries007/tfc/objects/entity/animal/EntityGrizzlyBearTFC.java:
```diff
@@ -0,0 +1,299 @@
+import net.dries007.tfc.world.classic.biomes.BiomesTFC;
+
+@ParametersAreNonnullByDefault
+public class EntityGrizzlyBearTFC extends EntityAnimalMammal implements IPredator, EntityAIStandAttack.IEntityStandAttack
```
This one should probably extend the other bear so you can cleanup copy-pasta below

Azaysol replied:
i had the black bear extend grizzly, since that one is the original brown bear

DisasterMoo commented on src/main/java/net/dries007/tfc/objects/entity/animal/EntityJackalTFC.java:
```diff
@@ -0,0 +1,207 @@
+import net.dries007.tfc.world.classic.biomes.BiomesTFC;
+
+@ParametersAreNonnullByDefault
+public class EntityJackalTFC extends EntityAnimalMammal implements IPredator
```
This one should probably extend hyena so you can cleanup copy-pasta below

Azaysol replied:
i had it extend coyote instead since they use the same model as the other extensions

DisasterMoo commented on src/main/java/net/dries007/tfc/types/DefaultRecipes.java:
```diff
@@ -390,10 +390,8 @@ public static void onRegisterHeatRecipeEvent(RegistryEvent.Register<HeatRecipe>
             new HeatRecipeSimple(IIngredient.of(ItemFoodTFC.get(Food.VENISON)), new ItemStack(ItemFoodTFC.get(Food.COOKED_VENISON)), 200, 480).setRegistryName("cooked_venison"),
             new HeatRecipeSimple(IIngredient.of(ItemFoodTFC.get(Food.RABBIT)), new ItemStack(ItemFoodTFC.get(Food.COOKED_RABBIT)), 200, 480).setRegistryName("cooked_rabbit"),
             new HeatRecipeSimple(IIngredient.of(ItemFoodTFC.get(Food.WOLF)), new ItemStack(ItemFoodTFC.get(Food.COOKED_WOLF)), 200, 480).setRegistryName("cooked_wolf"),
-            new HeatRecipeSimple(IIngredient.of(ItemFoodTFC.get(Food.CHEVON)), new ItemStack(ItemFoodTFC.get(Food.COOKED_CHEVON)), 200, 480).setRegistryName("cooked_chevon"),
```
So, there's no more chevon, duck and mongoose foods? Did you remove them from the lootables too? This is more of a concern for backwards compatibility, but I think that it will only delete them across inventories.

Azaysol replied:
changed on purpose, with all the animals being added it would have created a lot of new textures, so the meats were condensed to help with this. see the animal chart pinned in the discord chat for reference of the changes. the Lang file was changed to reflect how it should be done, but the items were kept the same to help with backwards compatibility. livestock birds all drop Poultry(chicken meat/texture) all huntable birds now drop Fowl( pheasent meat/texture) as an example

Azaysol replied:
the mongoose, re uses the hyena texture, and all dog/canine now drop the wolf meat/texture

DisasterMoo reviewed (changes requested):
A few concerns, see below

HeartyYF commented on src/main/resources/assets/tfc/lang/en_us.lang:
```diff
@@ -808,7 +808,7 @@ config.tfc.animals.donkey=Donkey
 config.tfc.animals.ocelot=Ocelot
 config.tfc.animals.wolf=Wolf
-config.tfc.animals.bear=Bear
+config.tfc.animals.bear=Grizzly Bear
```
It seems that this should be `config.tfc.animals.grizzly_bear`

Azaysol replied:
updated

HeartyYF commented:
Also, it seems that soundevent subtitles in this current lang file have some inconsistency with codes. Some "cry"s are missing from sounds.json and registered soundevents, and some subtitles that appear in codes are missing from the lang file.

eerussianguy commented:
> 
> 
> Also, it seems that soundevent subtitles in this current lang file have some inconsistency with codes. Some "cry"s are missing from sounds.json and registered soundevents, and some subtitles that appear in codes are missing from the lang file.

Not all of the animals have a full library of sounds. Some may only have 1 or 2

Azaysol commented:
> 
> 
> > Also, it seems that soundevent subtitles in this current lang file have some inconsistency with codes. Some "cry"s are missing from sounds.json and registered soundevents, and some subtitles that appear in codes are missing from the lang file.
> 
> Not all of the animals have a full library of sounds. Some may only have 1 or 2

I added a lot of new sounds, however the previous animals i added had 3 say and 3 cries. i had a hard time finding sounds for these animals so only added 3 says, and a couple cries for the predators since that seems to be used by them for the attack soundssound

Azaysol committed:
added some review request and added differences to some babies

Azaysol committed:
added more review request and fixed some rabbit stuff

DisasterMoo commented:
Leaving this here to write in changelog later:

```
New Animals:
- Direwolf: New predator
- Hare: New variation of rabbit
- Zebu: New variation of cow, livestock
- Boar: New huntable
- Gazelle: New huntable 
- Wildebeest: New huntable
- Quail: New livestock
- Grouse: New livestock
- Mongoose: New huntable
- Turkey: New huntable
- Jackal: New predator
- Musk Ox: New livestock
- Yak: New livestock
- Black Bear: New predator
- Coyote: New predator
- Cougar: New predator

Changes:
- Renamed Bear to Grizzly Bear
- Changed spawning locations to match variations (ie: Cow and Zebu)
- Meat drops are now less specific to a single animal and more grouped. Removed Hyena, Duck and Chevon meat items.
- Lots and lots of sound and texture changes.
```

If i missed something, please let me know.

Edit:
This PR closes #1357

Azaysol commented:
Lions were changes to hunt at Night like all the other Cats, also cougar is predator, not livestock lol
